nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -233,6 +248,20 @@ def test_getitem(self):
             assert_arrays_equal(sdict[-1][k], v[-1])
             assert_arrays_equal(sdict[[0, -1]][k], v[[0, -1]])
 
+    def test_extend(self):
+        total_nb_rows = DATA['tractogram'].streamlines.total_nb_rows
+        sdict = PerArraySequenceDict(total_nb_rows, DATA['data_per_point'])
```

Comment:
I'm not sure it will work as you intend or I don't get your point. There is a couple of differences between testing a `PerArrayDict` and a `PerArraySequenceDict`. For instance, the constructor of the first class takes the number of streamlines whereas the second takes the total number of points in a `ArraySequence`. Also, one checks `data_per_streamline` where the other `data_per_point`.